hubploy/auth.py

```diff
@@ -72,28 +82,33 @@ def registry_auth_aws(deployment, project, zone, service_key):
         raise FileNotFoundError(
             f'The service_key file {service_key_path} does not exist')
 
-    # move credentials to standard location
-    cred_dir = os.path.expanduser('~/.aws')
-    if not os.path.isdir(cred_dir):
-        os.mkdir(cred_dir)
-    shutil.copyfile(service_key_path, os.path.join(cred_dir, 'credentials'))
-
-    registry = f'{project}.dkr.ecr.{zone}.amazonaws.com'
-    # amazon-ecr-credential-helper installed in .circleci/config.yaml
-    # this adds necessary line to authenticate docker with ecr
-    docker_config_dir = os.path.expanduser('~/.docker')
-    os.makedirs(docker_config_dir, exist_ok=True)
-    docker_config = os.path.join(docker_config_dir, 'config.json')
-    if os.path.exists(docker_config):
-        with open(docker_config, 'r') as f:
-            config = json.load(f)
-    else:
-        config = {}
+    original_credential_file_loc = os.environ.get("AWS_SHARED_CREDENTIALS_FILE", "")
+
+    try:
+        # Set env variable for credential file location
+        os.environ["AWS_SHARED_CREDENTIALS_FILE"] = service_key_path
+
+        registry = f'{project}.dkr.ecr.{zone}.amazonaws.com'
+        # Requires amazon-ecr-credential-helper to already be installed
+        # this adds necessary line to authenticate docker with ecr
+        docker_config_dir = os.path.expanduser('~/.docker')
+        os.makedirs(docker_config_dir, exist_ok=True)
+        docker_config = os.path.join(docker_config_dir, 'config.json')
+        if os.path.exists(docker_config):
+            with open(docker_config, 'r') as f:
+                config = json.load(f)
+        else:
+            config = {}
+
+        config.setdefault('credHelpers', {})[registry] = 'ecr-login'
+        with open(docker_config, 'w') as f:
+            json.dump(config, f)
 
-    config.setdefault('credHelpers', {})[registry] = 'ecr-login'
-    with open(docker_config, 'w') as f:
-        json.dump(config, f)
+        yield
 
+    finally:
+        # Unset env variable for credential file location
+        unset_env_var("AWS_SHARED_CREDENTIALS_FILE", original_credential_file_loc)
```

Comment:
IMO this function is small enough that we can just write it out here:

```python
del os.environ["AWS_SHARED_CREDENTIALS_FILE"]
if original_credential_file_loc is not None:
	os.environ["AWS_SHARED_CREDENTIALS_FILE"] = original_credential_file_loc
```

Reply:
Well, I have to use that function two more times in `cluster_auth_aws()`, so I feel like it's easier to leave it as a function. I can change the logic to match this one, since deleting the value hubploy used seems like a good thing to do always.

Reply:
Fair!